Extension/package.json

```diff
@@ -3797,6 +3803,13 @@
           "group": "navigation"
         }
       ],
+      "editor/title": [
+        {
+          "command": "C_Cpp.BuildAndDebugFile",
+          "when": "(editorLangId == 'c' || editorLangId == 'cpp' || editorLangId == 'cuda-cpp') && C_Cpp.shortcutMenuBar.buildAndDebugFile",
```

Comment:
We didn't do this before, but is there an easy way to check if the active file is a header and disable the button by setting some context variable that could be checked here?

Reply:
We used the following to set `when` context variables from code, for FAR:

```
vscode.commands.executeCommand('setContext', 'cppReferenceTypes:hasResults', hasResults);
...
vscode.commands.executeCommand('setContext', 'refView.isGroupedByFile', this.referencesModel.groupByFile);
```

The value would need to be updated as the active file changes.  Not sure if our `isHeader` function is sufficient, though.  It looks like we have a better check in `getTasks` (which should probably be moved into `isHeader`).